### PR TITLE
feat: Add Thermo Fisher Qubit Flex adapter

### PR DESCRIPTION
**Pull request by Sivakumar36**
Add adapter for the Thermo Fisher Qubit Flex instrument. The adapter handles both XLSX and CSV exports of the instrument data.
The adapter uses [spectrophotometry](https://github.com/Benchling-Open-Source/allotropy/blob/main/src/allotropy/allotrope/schemas/adm/spectrophotometry/BENCHLING/2023/12/spectrophotometry.schema.json) schema of allotropy.

### CODE REVIEW

File: src/allotropy/parsers/thermo_fisher_qubit_flex/thermo_fisher_qubit_flex_parser.py
Comment (nathan-stender) [inline src/allotropy/parsers/thermo_fisher_qubit_flex/thermo_fisher_qubit_flex_parser.py:212]: This PR: https://github.com/Benchling-Open-Source/allotropy/pull/490 just added a constant for "N/A" (`from allotropy.parsers.constants import NOT_APPLICABLE`) , please use that.
```suggestion
                    device_identifier=NOT_APPLICABLE,
```

**Review by nathan-stender — CHANGES_REQUESTED** [review]

File: .idea/.gitignore
Comment (Sivakumar36) [reply .idea/.gitignore]: @nathan-stender  reverted the .idea directory

File: src/allotropy/parsers/thermo_fisher_qubit_flex/thermo_fisher_qubit_flex_parser.py
Comment (Sivakumar36) [reply src/allotropy/parsers/thermo_fisher_qubit_flex/thermo_fisher_qubit_flex_parser.py]: Hi @nathan-stender, comments fixed.

**Review by nathan-stender — DISMISSED** [review]

**Comment by nathan-stender** [comment]
@Sivakumar36 looks like you have some typing lint issues, you can check this by running `hatch run lint:typing` or just `hatch run lint`. I will re-approve after fixing!

**Comment by Sivakumar36** [comment]
@nathan-stender ,As @poojabalaji05 mentioned in here comments. 
 
Resolved the linting issues. Once the Qubit4 PR is approved and the code is merged, the test scripts for Qubit Flex should function correctly upon syncing the forks. This is because we've added the functionality to remove custom fields if their values are None in the Qubit4 PR. 
 
Also, noticed that the Tests failed because currently the parser's release state is not updated to 'recommended' which is expected and currently it is in 'working draft' and also this is not added to the readme.md file. I hope you will be updating this or else let me know we can also help on this.

**Comment by nathan-stender** [comment]
@Sivakumar36 for the README test, you will need to update the README in this PR for the test to pass, e.g.

under

```
### Candidate Release
  - Qiacuity dPCR
  - Mabtech Apex

### Working Draft
  - Thermo Fisher Qubit Flex
```

**Comment by nathan-stender** [comment]
@Sivakumar36 if you pull past this PR: https://github.com/Benchling-Open-Source/allotropy/pull/507, you can run `hatch run scripts:update-readme` to update the README file!

**Comment by Sivakumar36** [comment]
@nathan-stender I have updated the README.md file by adding the parser.
Also included the vendor in the NON_READY_PARSERS dictionary to avoid test failure in test_vendor_release_state function.
I noticed that the test script structure has been modified, so I have made the necessary adjustments from my end as well.
 
Additionally, the Qubit Flex test is failing due to null values being added to the custom information document dictionary. This issue will be resolved once we sync the fork after merging Qubit 4 PR, which removes the null keys.

**Comment by Sivakumar36** [comment]
Hi @nathan-stender,

I've pulled the latest changes from the main branch, committed the code, and ensured it passes all the checks. Awaiting your approval.

Thanks!

**Review by nathan-stender — APPROVED** [review]

**Review by nathan-stender — APPROVED** [review]

**Review by mdlu — APPROVED** [review]

**Review by nathan-stender — APPROVED** [review]

**Review by mdlu — APPROVED** [review]

**Pull request closed without merging**
